File: smallplayer/app/src/main/java/com/tonynowater/smallplayer/util/SPManager.java
package com.tonynowater.smallplayer.util;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by tony10532 on 2017/8/8.
 */
public class SPManager {

    public static final String SHARED_PREFERENCE_COMMON = "SHARED_PREFERENCE_COMMON";
    private static final String SP_KEY_LOGIN = "SP_KEY_LOGIN";
    private static final String SP_KEY_ACCESS_TOKEN = "SP_KEY_ACCESS_TOKEN";
    private static final String SP_KEY_REFRESH_TOKEN = "SP_KEY_REFRESH_TOKEN";
    private static final String SP_KEY_EXPIRES_IN = "SP_KEY_EXPIRES_IN";

    private static SPManager mInstance = null;

    private SharedPreferences sharedPreferences;

    public static SPManager getInstance(Context context) {
        if (mInstance == null) {
            mInstance = new SPManager(context);
        }
        return mInstance;
    }

    public SPManager(Context context) {
        sharedPreferences = context.getSharedPreferences(SHARED_PREFERENCE_COMMON, Context.MODE_PRIVATE);
    }

    public void setIsGoogleLogin(boolean login) {
        sharedPreferences.edit().putBoolean(SP_KEY_LOGIN, login).commit();
    }

    public boolean getIsGoogleLogin() {
        return sharedPreferences.getBoolean(SP_KEY_LOGIN, false);
    }

    public void setAccessToken(String token) {
        sharedPreferences.edit().putString(SP_KEY_ACCESS_TOKEN, token).commit();
    }

    public String getAccessToken() {
        return sharedPreferences.getString(SP_KEY_ACCESS_TOKEN, "");
    }

    public void setRefreshToken(String token) {
        sharedPreferences.edit().putString(SP_KEY_REFRESH_TOKEN, token).commit();
    }

    public String getRefreshToken() {
        return sharedPreferences.getString(SP_KEY_REFRESH_TOKEN, "");
    }

    public void setTokenExpireTime(long expires_in) {
        sharedPreferences.edit().putLong(SP_KEY_EXPIRES_IN, expires_in).commit();
    }

    public long getTokenExpireTime() {
        return sharedPreferences.getLong(SP_KEY_EXPIRES_IN, -1);
    }
}

File: smallplayer/app/src/main/java/com/tonynowater/smallplayer/util/SPManager.kt
package com.tonynowater.smallplayer.util

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by tony10532 on 2017/8/8.
 */
class SPManager(context: Context) {

    private val sharedPreferences: SharedPreferences

    var isGoogleLogin: Boolean
        get() = sharedPreferences.getBoolean(SP_KEY_LOGIN, false)
        set(login) {
            sharedPreferences.edit().putBoolean(SP_KEY_LOGIN, login).commit()
        }

    var accessToken: String
        get() = sharedPreferences.getString(SP_KEY_ACCESS_TOKEN, "")
        set(token) {
            sharedPreferences.edit().putString(SP_KEY_ACCESS_TOKEN, token).commit()
        }

    var refreshToken: String
        get() = sharedPreferences.getString(SP_KEY_REFRESH_TOKEN, "")
        set(token) {
            sharedPreferences.edit().putString(SP_KEY_REFRESH_TOKEN, token).commit()
        }

    var tokenExpireTime: Long
        get() = sharedPreferences.getLong(SP_KEY_EXPIRES_IN, -1)
        set(expires_in) {
            sharedPreferences.edit().putLong(SP_KEY_EXPIRES_IN, expires_in).commit()
        }

    var currentPlayAlbumId: Long
    get() = sharedPreferences.getLong(SP_KEY_CURRENT_PLAYING_ALBUM_ID, 1)
    set(albumId) {
        sharedPreferences.edit().putLong(SP_KEY_CURRENT_PLAYING_ALBUM_ID, albumId)
    }

    init {
        sharedPreferences = context.getSharedPreferences(SHARED_PREFERENCE_COMMON, Context.MODE_PRIVATE)
    }

    companion object {

        val SHARED_PREFERENCE_COMMON = "SHARED_PREFERENCE_COMMON"
        private val SP_KEY_LOGIN = "SP_KEY_LOGIN"
        private val SP_KEY_ACCESS_TOKEN = "SP_KEY_ACCESS_TOKEN"
        private val SP_KEY_REFRESH_TOKEN = "SP_KEY_REFRESH_TOKEN"
        private val SP_KEY_EXPIRES_IN = "SP_KEY_EXPIRES_IN"
        private val SP_KEY_CURRENT_PLAYING_ALBUM_ID = "SP_KEY_CURRENT_PLAYING_ALBUM_ID"

        private var mInstance: SPManager? = null

        fun getInstance(context: Context): SPManager {
            if (mInstance == null) {
                mInstance = SPManager(context)
            }
            return mInstance!!
        }
    }
}
